crates/symbolicator-service/src/metrics.rs

```diff
@@ -1,17 +1,351 @@
+
+            // send all the aggregated "counter like" metrics
+            for (AggregationKey { ty, name, tags }, value) in total_counters {
+                let _ = write!(
+                    &mut formatted_metric,
+                    "{prefix}{name}:{value}{ty}{formatted_global_tags}"
+                );
+
+                if let Some(tags) = tags {
+                    if formatted_global_tags.is_empty() {
+                        formatted_metric.push_str("|#");
+                    } else {
+                        formatted_metric.push(',');
+                    }
+                    formatted_metric.push_str(&tags);
+                }
+
+                let _ = sink.emit(&formatted_metric);
+
+                formatted_metric.clear();
+            }
+
+            // send all the aggregated "distribution like" metrics
+            // we do this in a batched manner, as we do not actually *aggregate* them,
+            // but still send each value individually.
+            for (AggregationKey { ty, name, tags }, value) in total_distributions {
+                suffix.push_str(&formatted_global_tags);
+
+                if let Some(tags) = tags {
+                    if formatted_global_tags.is_empty() {
+                        suffix.push_str("|#");
+                    } else {
+                        suffix.push(',');
+                    }
+                    suffix.push_str(&tags);
+                }
+
+                for batch in value.chunks(DISTRIBUTION_BATCH_SIZE) {
+                    formatted_metric.push_str(&prefix);
+                    formatted_metric.push_str(name);
+
+                    for value in batch {
+                        let _ = write!(&mut formatted_metric, ":{value}");
+                    }
+
+                    formatted_metric.push_str(ty);
+                    formatted_metric.push_str(&suffix);
+
+                    let _ = sink.emit(&formatted_metric);
+                    formatted_metric.clear();
+                }
+
+                suffix.clear();
+            }
+        }
+    };
+
+    thread::Builder::new()
+        .name("metrics-aggregator".into())
+        .spawn(thread_fn)
+        .unwrap();
+
+    local_aggregators
+}
+
+fn aggregate_all(aggregators: &LocalAggregators) -> (AggregatedCounters, AggregatedDistributions) {
+    let mut total_counters = AggregatedCounters::default();
+    let mut total_distributions = AggregatedDistributions::default();
+
+    for local_aggregator in aggregators.iter() {
+        let (local_counters, local_distributions) = {
+            let mut local_aggregator = local_aggregator.lock().unwrap();
+            (
+                std::mem::take(&mut local_aggregator.aggregated_counters),
+                std::mem::take(&mut local_aggregator.aggregated_distributions),
+            )
+        };
+
+        // aggregate all the "counter like" metrics
+        if total_counters.is_empty() {
```

Comment:
The same remark from before still applies: `total_counters` is definitely empty here.

Reply:
Well yes, on the first iteration it is. This is pretty much a manual implementation of "take on first iteration, extend on all the rest"

Reply:
Yeah I missed the loop, my bad.